firestore/bulkwriter.go

```diff
@@ -0,0 +1,521 @@
+package firestore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	vkit "cloud.google.com/go/firestore/apiv1"
+	pb "google.golang.org/genproto/googleapis/firestore/v1"
+)
+
+const (
+	// MaxBatchSize is the max number of writes to send in a request
+	MaxBatchSize = 20
```

Comment:
I think that we want to make sure that clients are aware of these limitations, but they are unlikely to use them. This is probably better handled in the BulkWriter struct/class documentation rather than as exposed consts. I'll hide them.